https://trac.macports.org/ticket/64115
OpenSSL 1.1 include path needs to be specified with higher precedence than ${prefix}/include
since openssl shim port now makes OpenSSL 3 include headers accessible from ${prefix}/include

https://trac.macports.org/ticket/64259
Also we need to use the openssl executable from the same path

diff --git a/cmake/ssl.cmake b/cmake/ssl.cmake
index 293281cf1..96a48d06186 100644
--- a/cmake/ssl.cmake
+++ b/cmake/ssl.cmake
@@ -236,7 +236,9 @@
       )
       SET(OPENSSL_FOUND TRUE)
       FIND_PROGRAM(OPENSSL_EXECUTABLE openssl
-        DOC "path to the openssl executable")
+        DOC "path to the openssl executable"
+        PATHS ${OPENSSL_ROOT_DIR}/bin NO_DEFAULT_PATH)
+      MESSAGE(STATUS "OpenSSL Executable - ${OPENSSL_EXECUTABLE}")
       IF(OPENSSL_EXECUTABLE)
         SET(OPENSSL_EXECUTABLE_HAS_ZLIB 0)
         EXECUTE_PROCESS(
@@ -313,7 +315,7 @@
         SET(SSL_LIBRARIES ${SSL_LIBRARIES} ${LIBDL})
       ENDIF()
       MESSAGE(STATUS "SSL_LIBRARIES = ${SSL_LIBRARIES}")
-      INCLUDE_DIRECTORIES(SYSTEM ${OPENSSL_INCLUDE_DIR})
+      INCLUDE_DIRECTORIES(BEFORE ${OPENSSL_INCLUDE_DIR})
     ELSE()
       RESET_SSL_VARIABLES()
       FATAL_SSL_NOT_FOUND_ERROR(
